File: src/ObjectiveC/ListDemo/ListDemo/ViewController.m
//
//  ViewController.m
//  ListDemo
//
//  Created by Sergio Cirasa on 1/31/16.
//  Copyright © 2016 Sergio Cirasa. All rights reserved.
//

#import "ViewController.h"
#import "ElasticFlowLayout.h"

@interface ViewController () <UICollectionViewDataSource, UICollectionViewDelegate, UICollectionViewDelegateFlowLayout>

@end

@implementation ViewController

//------------------------------------------------------------------------------------------------------------------------------
- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.collectionView.collectionViewLayout = [[ElasticFlowLayout alloc] init];
    [self.collectionView registerClass:[UICollectionViewCell class] forCellWithReuseIdentifier:@"Cell"];
    self.collectionView.delegate = self;
    self.collectionView.dataSource = self;
}
//------------------------------------------------------------------------------------------------------------------------------
#pragma mark - UICollectionView DataSource & Delegate
-(NSInteger)numberOfSectionsInCollectionView:(UICollectionView *)collectionView
{
    return 1;
}
//------------------------------------------------------------------------------------------------------------------------------
-(NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section
{
    return 100;
}
//------------------------------------------------------------------------------------------------------------------------------
-(UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath
{
    UICollectionViewCell *cell = (UICollectionViewCell *)[collectionView dequeueReusableCellWithReuseIdentifier:@"Cell" forIndexPath:indexPath];
    cell.backgroundColor = [UIColor colorWithWhite:1.0 alpha:1.0];
    
    return cell;
}
//------------------------------------------------------------------------------------------------------------------------------
- (CGSize)collectionView:(UICollectionView *)collectionView layout:(UICollectionViewLayout*)collectionViewLayout sizeForItemAtIndexPath:(NSIndexPath *)indexPath
{
    CGFloat padding = 10;
    CGFloat cellHeight = 50;
    return CGSizeMake(self.view.bounds.size.width - (padding*2), cellHeight);
}
//------------------------------------------------------------------------------------------------------------------------------
@end

File: src/Swift/ElasticList/ElasticList/ViewController.swift
//
//  ViewController.swift
//  ElasticList
//
//  Created by Sergio Cirasa on 1/31/16.
//  Copyright © 2016 Sergio Cirasa. All rights reserved.
//

import UIKit

class ViewController: UICollectionViewController{
    
    final let PADDING = CGFloat(10)
    final let CELL_HEIGHT = CGFloat(50.0)
    final let CELL_IDENTIFIER = "Cell"
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.collectionView!.collectionViewLayout = ElasticFlowLayout();
        self.collectionView!.delegate = self;
        self.collectionView!.dataSource = self;
    }
    //-----------------------------------------------------------------------------------------------------------------------------
    // MARK: - UICollectionViewDataSource
    override func numberOfSectionsInCollectionView(collectionView: UICollectionView) -> Int{
        return 1;
    }

    override func collectionView(collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int{
        return 100;
    }
    
    override func collectionView(collectionView: UICollectionView, cellForItemAtIndexPath indexPath: NSIndexPath) -> UICollectionViewCell{
        let cell = collectionView.dequeueReusableCellWithReuseIdentifier("Cell", forIndexPath: indexPath) as! CustomCollectionViewCell
        cell.textLabel?.text = NSString(format:"Cell %d",indexPath.row) as String
         cell.backgroundColor = UIColor.whiteColor()
        return cell;
    }
    //-----------------------------------------------------------------------------------------------------------------------------
    // MARK: - UICollectionViewDelegate
     func collectionView(collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout,sizeForItemAtIndexPath indexPath: NSIndexPath) -> CGSize{
        
        return CGSizeMake((CGFloat)(self.view.bounds.width - (PADDING*2)), CELL_HEIGHT);
    }
    
}